dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/insert/validation/ProgramInstanceRepeatableStageCheck.java

```diff
@@ -73,20 +75,20 @@ public ImportSummary check(ImmutableEvent event, WorkContext ctx )
         return success();
     }
 
-    private boolean hasProgramStageInstance( JdbcTemplate jdbcTemplate, String programStageUid )
+    private boolean hasProgramStageInstance( JdbcTemplate jdbcTemplate, long programStageId, long trackedEntityInstanceId )
     {
         // @formatter:off
         final String sql = "select exists( " +
                 "select * " +
                 "from programstageinstance psi " +
                 "  join programinstance pi on psi.programinstanceid = pi.programinstanceid " +
-                "  join programstage ps on psi.programstageid = ps.programstageid " +
```

Comment:
Switched to using primary key (foreign key) column instead of uid and hence this join is no longer required.